src/common/string_util.cc

```diff
@@ -276,6 +276,71 @@ std::pair<std::string, std::string> SplitGlob(std::string_view glob) {
   return {prefix, ""};
 }
 
+StatusOr<std::vector<std::string>> SplitArguments(std::string_view in) {
+  std::vector<std::string> arguments;
+  std::string current_string;
+
+  enum { NORMAL, QUOTED, SINGLE_QUOTED, ESCAPE } state = NORMAL;
+
+  bool is_quoted = false;
+  for (const char c : in) {
+    switch (state) {
+      case NORMAL:
+        if (std::isspace(c)) {
+          if (!current_string.empty()) {
+            arguments.emplace_back(std::move(current_string));
+            current_string.clear();
+          }
+          // skip spaces
+        } else if (c == '"' || c == '\'') {
+          state = c == '"' ? QUOTED : SINGLE_QUOTED;
+          is_quoted = c == '"';
```

Comment:
It seems `state` itself is enough to determine whether it is quoted? Is `is_quoted` necessary?

Reply:
Ahh I got it. Maybe we can use `state_before_escape` to store the previous before ESCAPE and revert to it after ESCAPE.